howtos/how-to-make-clickable-tabs-for-conversations/index.html

```diff
@@ -0,0 +1,150 @@
+    const me = new Talk.User({
+      id: "tabConversationExampleSupportAgent",
+      name: "Alice",
+      email: "alice@example.com",
+      role: "default",
+      photoUrl: "https://talkjs.com/images/avatar-1.jpg",
+      welcomeMessage: "Hey there! How can I help?",
+    });
+
+    const talkSession = new Talk.Session({
+      appId: "<APP_ID>", // replace with your app ID
+      me: me,
+    });
+
+    chatbox = talkSession.createChatbox();
+    chatbox.mount(document.getElementById("talkjs-container"));
+
+    chatbox.onCustomMessageAction("openTab", (event) => {
+    //The handler for our open tab custom action. This event
+    //creates the tab for the user and swaps to a conversation with the selected user.
+      const messageSender = new Talk.User({
+          id: event.message.senderId,
+          name: event.message.sender.name,
+          email: event.message.sender.email,
+          role: event.message.sender.role,
+          photoUrl: event.message.sender.photoUrl,
+          welcomeMessage: event.message.sender.welcomeMessage,
+      });
+      
+      const conversation = talkSession.getOrCreateConversation(
+          event.message.senderId
+      );
+    
+      convId = conversation.id;
+      tab = document.getElementById(convId);
+      if(tab){
+        tab.click();
+      }
+      else{
+        conversation.setParticipant(me);
+        conversation.setParticipant(messageSender);
+
+        const groupTab = document.getElementById('tabConversationExample1')
+        if(!groupTab){
+          createTab('tabConversationExample1', 'Group');
+        }
+
+        const privateConversation = document.getElementById(convId);
+        if(!privateConversation){
+          createTab(convId, event.message.sender.name);
+        }
+
+        tab = document.getElementById(convId);
+        tab.click();
+      }
+    });
+  });
+
+  function createTab(tabId, tabName) {
+    // Create the new tab button
+    var btn = document.createElement("button");
+    btn.id = tabId;
+    btn.className = "tablinks";
+    btn.textContent = tabName;
+    btn.onclick = function(event) { openTab(event, tabId, tabName); };
+
+    // Add the new tab button to the tab container
+    var tabContainer = document.getElementById("dynamicTabContainer");
+    tabContainer.appendChild(btn);
+  }
+
+  function openTab(event, tabId, tabName) {
+    var i, tablinks;
+
+    // Get all elements with class="tablinks" and remove the class "active"
+    tablinks = document.getElementsByClassName("tablinks");
+    for (i = 0; i < tablinks.length; i++) {
+        tablinks[i].className = tablinks[i].className.replace(" active", "");
+    }
+
+    // Show the current tab, and add an "active" class to the button that opened the tab
+    document.getElementById(tabId).style.display = "block";
+    event.currentTarget.className += " active";
+    
+    //changes the active conversation in the chatbox to the tab/conversationId we have selected.
+    chatbox.select(tabId);
```

Comment:
Passing a `string` to `chatbox.select()` means that TalkJS will not synchronize data and will instead assume a conversation with ID `tabId` to already exists. What we want is to create a new conversation with `session.getConversation()` and set the participants, which is what you've done in the code.

However, creating the conversation & setting the participant only happens when you pass the `Conversation` object you got from `session.getConversation()` to `chatbox.select()`. So, pass `Conversation` instead of `string` to `chatbox.select()` here to make sure the conversation gets created properly.